### ISSUE TITLE
NPE when writing extra data records with no data

### ISSUE THREAD

**Issue by srikanth-lingala** [srikanth-lingala/zip4j#154]
@srikanth-lingala @nicolabeghin  
I have other problem 。when I  add fille in zip ,will get NullPointException. but not All zip . 
reappear it  steps:

1. download this file  [http://oss.youxifan.com/MGInc6E](url) .yes it's  apk .(apk is a zip)
2. use the add Api code .
```
    ZipFile zipFile = new ZipFile("/Users/engine/apk/test.apk");
    ArrayList<File> addFiles = new ArrayList<>();
    addFiles.add(new File("/Users/engine/apk/test.txt"));
    ZipParameters parameters = new ZipParameters();
    parameters.setCompressionMethod(CompressionMethod.DEFLATE);
    parameters.setCompressionLevel(CompressionLevel.NORMAL);
    // 目标路径
    parameters.setRootFolderNameInZip("META-INF/");
    zipFile.addFiles(addFiles, parameters);
```
3.  you will get  java.lang.NullPointerException and  damage the apk
```
Stack trace:
 net.lingala.zip4j.exception.ZipException: java.lang.NullPointerException
	at net.lingala.zip4j.headers.HeaderWriter.writeFileHeader(HeaderWriter.java:501)
	at net.lingala.zip4j.headers.HeaderWriter.writeCentralDirectory(HeaderWriter.java:382)
	at net.lingala.zip4j.headers.HeaderWriter.finalizeZipFile(HeaderWriter.java:177)
Caused by: java.lang.NullPointerException
	at java.io.OutputStream.write(OutputStream.java:75)
	at net.lingala.zip4j.headers.HeaderWriter.writeRemainingExtraDataRecordsIfPresent(HeaderWriter.java:54
	at net.lingala.zip4j.headers.HeaderWriter.writeFileHeader(HeaderWriter.java:495)
	... 37 more
```

4.  I have fix this Exception  by modify  HeaderWriter.writeRemainingExtraDataRecordsIfPresent() .at the latest code :
```
    if(extraDataRecord.getData()!=null){
        outputStream.write(extraDataRecord.getData());
      }
```
but  I'm not sure I'm right

_Originally posted by @lovekill in https://github.com/srikanth-lingala/zip4j/issues/148#issuecomment-595650771_

**Comment by srikanth-lingala** [comment]
Issue fixed, will include it in next release

**Comment by srikanth-lingala** [comment]
Fix included in v2.5.0 released today. A couple of more features have been added in this release. Please have a look at [release notes][1].

[1]: https://github.com/srikanth-lingala/zip4j/releases
